src/Libraries/CoreNodeModels/CustomSelectionItem.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.ComponentModel;
+using System.Windows.Input;
+using Newtonsoft.Json;
+
+namespace CoreNodeModels
+{
+    /// <summary>
+    /// Represents an Enum item
+    /// </summary>
+    public class CustomSelectionItem
+    {
+        public string Name { get; set; }
+
+        public string Value { get; set; }
+
+        public CustomSelectionItem()
+        {
+
+        }
+    }
+
+
+    public class CustomSelectionItemViewModel : INotifyPropertyChanged, IDataErrorInfo
+    {
+        private readonly CustomSelectionItem customSelectionItem;
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        #region Properties
+
+        [JsonIgnore]
+        public Func<CustomSelectionItem, bool> IsUnique { get; set; }
+
+        [JsonIgnore]
+        public Action ItemChanged { get; set; }
+
+        [JsonIgnore]
+        public Action<CustomSelectionItemViewModel> RemoveRequested { get; set; }
+
+        public string Name
+        {
+            get
+            {
+                return customSelectionItem.Name;
+            }
+            set
+            {
+                customSelectionItem.Name = value;
+
+                OnPropertyChanged("Name");
```

Comment:
use `nameof`

Reply:
OnPropertyChanged(nameof(customSelectionItem.Name))

You mean this?

Reply:
yes, leverage the compiler to avoid typos and name changes.